vendors/nordic/boards/nrf52840-dk/ports/ble/iot_ble_hal_gap.c

```diff
@@ -850,8 +850,26 @@ ret_code_t prvBTAdvDataConvert( ble_advdata_t * xAdvData,
     {
         /* The first two UUID go to the initial advertisement packet */
         xAdvData->uuids_complete = xCompleteUUIDS;
-        xAdvData->p_slave_conn_int->max_conn_interval = pxParams->ulMaxInterval;
-        xAdvData->p_slave_conn_int->min_conn_interval = pxParams->ulMinInterval;
+
+        if( ( pxParams->ulMaxInterval != 0 ) && ( pxParams->ulMinInterval != 0 ) )
+        {
```

Comment:
if both values (ulMaxInterval and ulMinInterval) are 0's, both the if and the else will not be executed.
(not sure if it is a possibility)

Reply:
if both values are zeros, it means the connection interval won't be included in advertisement, so its a no-action.

Reply:
I'd have an else branch and call it out it is intended to do nothing...

Reply:
Good callout. Added an empty else.